block/store.go

```diff
@@ -52,22 +53,46 @@ func StoreIncoming(packedBlock []byte, performRescan rescanType) error {
 	// extract incoming block record, checking for correct sequence
 	header, digest, data, err := blockrecord.ExtractHeader(packedBlock, height+1)
 	if nil != err {
+		globalData.log.Errorf("extract header error: %s", err)
 		return err
 	}
 
-	// ensure correct linkage
-	if previousBlock != header.PreviousBlock {
-		return fault.ErrPreviousBlockDigestDoesNotMatch
+	if blockrecord.IsDifficultyAppliedVersion(header.Version) {
+		if difficulty.IsAdjustBlock(header.Number) {
+			nextDifficulty, prevDifficulty, err := blockrecord.AdjustDifficultyAtBlock(header.Number)
+			// if any error happens for storing block, reset difficulty back to old value
+			defer func() {
```

Comment:
Just is my suggestion. It's fine now but you can consider to change if needed. 
You define defer func as anonymous function so it references to the outside function (the caller function), so that when the defer function is called, it will use `prevDifficulty` referenced from the caller and if the  `prevDifficulty` is changed before, defer function will use the changed value, it maybe causes some unexpected problems. 
The better way is pass the `prevDifficulty` into the defer function instead of using it as a reference from outside function.

Reply:
It will act as same, but I will put it as variable to avoid some unexpected change